src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -84,6 +84,31 @@ public boolean isSameAppointment(Appointment otherAppointment) {
                 && otherAppointment.timePeriod.equals(timePeriod);
     }
 
+    /**
+     * Returns true if both appointments has overlapping time period.
+     * Overlapping is defined as within, the same or partly coinciding timePeriods.
+     */
+    public boolean hasOverlappingTimePeriod(Appointment otherAppt) {
+        Time existingStartTime = this.getStartTime();
+        Time existingEndTime = this.getEndTime();
+        Time otherStartTime = otherAppt.getStartTime();
+        Time otherEndTime = otherAppt.getEndTime();
+
+        // other startTime is before existing startTime &&
+        // other endTime is before or equal to existing startTime --> return false
+        if (otherStartTime.compareTo(existingStartTime) < 0 && otherEndTime.compareTo(existingStartTime) <= 0) {
```

Comment:
Wondering whether we can have a function in `TimePeriod` to do these checks for overlap between TimePeriods and we can just call that here?

Reply:
Good idea! I will add isBefore, isAfter methods etc to the TimePeriod class.